logback-ecs-encoder/src/main/java/co/elastic/logging/logback/EcsEncoder.java

```diff
@@ -113,10 +115,12 @@ public void close() throws IOException {
             }
         }
         IThrowableProxy throwableProxy = event.getThrowableProxy();
-        if (throwableProxy instanceof ThrowableProxy) {
-            EcsJsonSerializer.serializeException(builder, ((ThrowableProxy) throwableProxy).getThrowable(), stackTraceAsArray);
-        } else if (throwableProxy != null) {
-            EcsJsonSerializer.serializeException(builder, throwableProxy.getClassName(), throwableProxy.getMessage(), throwableProxyConverter.convert(event), stackTraceAsArray);
+        if(throwableProxy != null) {
+            if (throwableConverter != null) {
+                EcsJsonSerializer.serializeException(builder, throwableProxy.getClassName(), throwableProxy.getMessage(), throwableConverter.convert(event), stackTraceAsArray);
+            } else {
```

Comment:
There needs to be a 3rd case when `throwableProxy` is not an instance of `ThrowableProxy` - serialize with `ThrowableProxyConverter` in that case. You'll need two fields then: one `ThrowableProxyConverter` (not user-definable) and `ThrowableHandlingConverter` (user-definable).

Reply:
🤔  is there a way to have a test that produces a logging event with a `IThrowableProxy` that is not of type `ThrowableProxy`? It would be good to have test coverage for this scenario so someone doesn't remove that logic in the future (like I just did)

Reply:
Good question. I think that case happens when using async appenders.

Reply:
Thanks for the tip. Will take a look

Reply:
Tested it with an async appender and it still used a `ThrowableProxy` so for know I have just added back in that code though without any additional tests